internal/distributedtracing/distributedtracing.go

```diff
@@ -0,0 +1,277 @@
+package distributedtracing
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/open-policy-agent/opa/config"
+	"github.com/open-policy-agent/opa/logging"
+	"github.com/open-policy-agent/opa/util"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	defaultEnable               = false
+	defaultAddress              = "localhost:4317"
+	defaultServiceName          = "opa"
+	defaultSampleRatePercentage = int(100)
+	defaultEncyrptionScheme     = "off"
+	defaultEncryptionSkipVerify = false
+	defaultTLSCertFile          = ""
+	defaultTLSPrivateKeyFile    = ""
+	defaultTLSCACertFile        = ""
+)
+
+var supportedEncryptionScheme = map[string]struct{}{
+	"off": {}, "tls": {}, "mtls": {},
+}
+
+func isSupportedEncryptionScheme(scheme string) bool {
+	_, ok := supportedEncryptionScheme[scheme]
+	return ok
+}
+
+func isSupportedSampleRatePercentage(sampleRate int) bool {
+	return sampleRate >= 0 && sampleRate <= 100
+}
+
+type distributedTracingConfig struct {
+	Enabled               *bool   `json:"enabled,omitempty"`
+	Address               *string `json:"address,omitempty"`
+	ServiceName           *string `json:"service_name,omitempty"`
+	SampleRatePercentage  *int    `json:"sample_percentage,omitempty"`
+	EncryptionScheme      *string `json:"encryption,omitempty"`
+	EncryptionSkipVerify  *bool   `json:"allow_insecure_tls,omitempty"`
+	TLSCertFile           *string `json:"tls_cert_file,omitempty"`
+	TLSCertPrivateKeyFile *string `json:"tls_private_key_file,omitempty"`
+	TLSCACertFile         *string `json:"tls_ca_cert_file,omitempty"`
```

Comment:
For string values, we should only need `*string` when we **need to differentiate** `""` and _not set_. Is that the case here? At least for TLS, I doubt it -- empty string isn't a valid default.

Reply:
Good point, empty string shouldn't be valid for any of these